Src/API/Action.lua

```diff
@@ -217,6 +235,15 @@ export type Action = typeof(Action.Prototype) & ActionSettings
 export type ActionSettings = {
 	Name: string,
 	Uuid: string,
+
+	Arguments: {
+		{
+			ArgumentName: string,
+			ArgumentType: string,
+			ArgumentDefault: any?,
+			ArgumentIsOptional: boolean?,
```

Comment:
can we just do Name, Type, Default, IsOptional?

Reply:
Sure.

Reply:
Another thought: Right now the types I have as acceptible for Actions on the server are string, number, boolean and player if you can take that type and deny it if it isnt any of those. One other thing to thing about is using the terminology `required` vs. `optional`

```
name: "Fruit"
type: "string"
default: "banana"
required: false
```